### DISCUSSION TITLE
Is it possible to auto select poster files?

### DISCUSSION THREAD

**Discussion by AwesomeAustn** [mza921/Plex-Auto-Collections#164] [Q&A] [answered]
Is it possible to auto select poster files? It will help when running multiple collections at once. For me, it stops and asks for every collection:

image_server:
    poster_directory: ../config/posters
    background_directory: ../config/backgrounds
    image_directory: ../config/images

| Locating Image Server...
| Using /home/austin/PlexAutoCollections/config/posters for Posters Directory
| Using /home/austin/PlexAutoCollections/config/backgrounds for Backgrounds Directory
| Using /home/austin/PlexAutoCollections/config/images for Images Directory

| Detail: sort_title updated to *150
| Detail: collection_mode updated to default
| Detail: collection_order updated to alpha
| background not found at: /home/austin/PlexAutoCollections/config/backgrounds/Animated Kids Popular.*
| background not found at: /home/austin/PlexAutoCollections/config/images/Animated Kids Popular.*
| 0) Do Nothing
| 1) [url] Animated Kids Popular.png
| 2) [file] /home/austin/PlexAutoCollections/config/posters/Animated Kids Popular.png
| Choose poster number: 2
| Detail: poster_directory updated poster to [file] /home/austin/PlexAutoCollections/config/posters/Animated Kids Popular.png

**Comment by ramair02** [comment]
How do you have your posters organized? How about your image_server config?

I have one folder for all of the collections posters and they're named [Collection].ext. My script only asks which poster if I have two, e.g. Rocky.jpg & Rocky.png.

If you're getting this on every collection, I suspect you either have multiple files or you have both a folder for each collection with poster and a general folder with all collection posters.

**Reply by AwesomeAustn** [reply]
My posters are named the title of the collection, and I only have one of each. I put them all in the poster directory without collection folders.

image_server:
    poster_directory: ../config/posters
    background_directory: ../config/backgrounds
    image_directory: ../config/images

**Comment by ramair02** [comment]
Hmm, I'm not sure then. Best I can do is share how mine is setup:

```
image_server:
  poster_directory: B:\Plex Collections Metadata
  background_directory: ../config/backgrounds
  image_directory: B:\Plex Collections Metadata
```

I don't use backgrounds and I don't have any folders in the main poster directory, so the background_directory and image_directory lines don't do anything. Maybe try removing the image_directory line from your script? I believe that's really for poster.ext within folders of the Collection Name. 

Here's a quick screengrab sampling my folder:
![image](https://user-images.githubusercontent.com/5008354/103569654-afb68480-4e95-11eb-9f61-95d12fcb94c0.png)

And here's how it runs with no interaction from me:
![image](https://user-images.githubusercontent.com/5008354/103569910-25225500-4e96-11eb-9059-2d8935ff8f80.png)

**Reply by AwesomeAustn** [reply]
I tried removing the Background and Image paths, and it's still asking.

image_server:
    poster_directory: ../config/posters

| Locating Image Server...
| Using /home/austin/PlexAutoCollections/config/posters for Posters Directory
| Backgrounds Directory was blank and defaults were not found
| Images Directory was blank and defaults were not found

| Detail: sort_title updated to *401
| Detail: collection_mode updated to default
| Detail: collection_order updated to alpha
| 0) Do Nothing
| 1) [url] Mindfuck by LISH.png
| 2) [file] /home/austin/PlexAutoCollections/config/posters/Mindfuck by LISH.png
| Choose poster number: 2
| Detail: poster_directory updated poster to [file] /home/austin/PlexAutoCollections/config/posters/Mindfuck by LISH.png

**Reply by ramair02** [reply]
Ok, I'm all out of ideas! Sorry I couldn't help. I'm sure someone more knowledgeable than me will know what's going on.

It must have something to do with [url] vs. [file]. Perhaps something related to your OS or containers or some part of your system being remote? I'm on Windows and my system acts as both a server and NAS. So everything is local. I've never seen the script ask between [url] and [file].

EDIT: Actually, I do see [url] when I run the script, but only when it is pulling artwork from TMDB (when it doesn't find a corresponding file in my folder). And this is all automated, no user interaction.

If you run the script with the --update flag, does it pick one of the posters? Or does it not add any artwork at all?

**Reply by AwesomeAustn** [reply]
Thanks for your input and effort!

**Reply by BlueSkillZ99** [reply]
could you post the config for the collection part? are you using any commands that pull posters from TMDb, e.g tmdb_id, tmdb_actor or other URL? There are several. Do you start the script with or without  -u or —update? 

However, I do not quite understand, why it only shows the filename for option 1. I am seeing the full URL when I run e.g tmdb_actor plus local file without —update

**Reply by AwesomeAustn** [reply]
https://github.com/mza921/Plex-Auto-Collections/files/5767498/configMovie.yml.txt
https://github.com/mza921/Plex-Auto-Collections/files/5767499/configTV.yml.txt

python3 plex_auto_collections.py --config-path ../config/config.yml

I just copy a collection to the main config and change the Plex settings to either Movie and movie, or TV Shows and show.

**Reply by BlueSkillZ99** [reply]
I think you are seeing exactly the intended behaviour, as described in the readme. poster specifies a url, which creates a second option. you are not turning off user interaction by including --update, so you get asked to pick

https://github.com/mza921/Plex-Auto-Collections/blob/master/README.md#poster-collection-attribute

1. get rid of all the „poster: xxxx.png“ rows in your config, they are obsolete. You need that for URLs, not local files. If local filename/location differs from collection name use „file_poster:„ . But if the poster are in the default folder and file and collection names match you do not need to specify anything
2. run the py with —update. Then it will always pick option 1 without user interaction

**Reply by AwesomeAustn** [reply]
That did it! I removed the poster row. Thank you so much!

**Comment by AwesomeAustn** [comment, marked as answer]
BlueSkillZ99
https://github.com/mza921/Plex-Auto-Collections/discussions/164#discussioncomment-265973
